twoliter/src/docker/container.rs

```diff
@@ -0,0 +1,59 @@
+use crate::common::exec;
+use anyhow::Result;
+use log::{debug, warn};
+use std::path::Path;
+use tokio::process::Command;
+
+pub(crate) struct DockerContainer {
+    name: String,
+}
+
+impl DockerContainer {
+    /// Create a docker image with the given name from the image by using `docker create`.
+    pub(crate) async fn new<S1, S2>(container_name: S1, image: S2) -> Result<Self>
+    where
+        S1: Into<String>,
+        S2: Into<String>,
+    {
+        let name = container_name.into();
+        let image = image.into();
+
+        // Make sure previous versions of this container are stopped deleted.
+        let args = vec!["stop".to_string(), name.to_string()];
+        if let Err(e) = exec(Command::new("docker").args(args), true).await {
+            warn!("Unable to stop container '{name}': {e}")
+        }
+        let args = vec!["rm".to_string(), name.to_string()];
+        if let Err(e) = exec(Command::new("docker").args(args), true).await {
+            warn!("Unable to remove container '{name}': {e}")
+        }
+
+        debug!("Creating docker container '{name}' from image '{image}'");
+
+        // Create the new container.
+        let args = vec![
+            "create".to_string(),
+            "--rm".to_string(),
```

Comment:
When is the container removed? I'm used to seeing this with `docker run` where the container is removed when the process stops.

Reply:
Added implementation for drop.